File: frontend/types/property.ts
export type PropertyStatus = "available" | "sold" | "rented";

export interface Property {
  id: string;
  title: string;
  description: string;
  price: number;
  location: string;
  bedrooms: number;
  bathrooms: number;
  area_sqm: number;
  built_year: number | null;
  subway_distance_m: number | null;
  tags: string[];
  images: string[];
  status: PropertyStatus;
  created_at: string;
}

export interface PropertyListResponse {
  items: Property[];
  total: number;
  page: number;
  page_size: number;
}

File: backend/app/schemas/property.py
from datetime import datetime
from enum import Enum

from pydantic import BaseModel, Field


class PropertyStatus(str, Enum):
    AVAILABLE = "available"
    SOLD = "sold"
    RENTED = "rented"


class Property(BaseModel):
    id: str
    title: str
    description: str
    price: float
    location: str
    bedrooms: int
    bathrooms: int
    area_sqm: float
    built_year: int | None = None
    subway_distance_m: int | None = None
    tags: list[str] = Field(default_factory=list)
    images: list[str]
    status: PropertyStatus
    created_at: datetime


class PropertyListResponse(BaseModel):
    items: list[Property]
    total: int
    page: int
    page_size: int
